src/internal/value/value.go

```diff
@@ -0,0 +1,272 @@
+	// Schema Schema
+	// REVIEW: Should we guard against?
+	// FileSizeLimit
+	// MaximumYAMLDepth
+	// Timeout
+}
+
+// ParseFiles parses the given files in order, overwriting previous values with later values, and returns a merged
+// Values map.
+func ParseFiles(ctx context.Context, paths []string, _ ParseFilesOptions) (_ Values, err error) {
+	m := make(Values)
+	start := time.Now()
+	defer func() {
+		logger.From(ctx).Debug("values parsing complete",
+			"duration", time.Since(start),
+			"files", len(paths))
+	}()
+
+	// No files given
+	if len(paths) <= 0 {
+		return Values{}, nil
+	}
+	// Validate file extensions
+	for _, path := range paths {
+		ext := strings.ToLower(filepath.Ext(path))
+		if ext != ".yaml" && ext != ".yml" {
+			return nil, &InvalidFileExtError{FilePath: path, Ext: ext}
+		}
+	}
+
+	logger.From(ctx).Debug("parsing values files", "paths", paths)
+	for _, path := range paths {
+		// Allow for cancellation
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+			var vals Values
+			if helpers.IsURL(path) {
+				return nil, fmt.Errorf("remote values files not yet supported, url=%s", path)
+			}
+			_, err := helpers.IsTextFile(path)
+			if err != nil {
+				return nil, err
+			}
+			vals, err = parseLocalFile(ctx, path)
+			if err != nil {
+				return nil, err
+			}
+			// Done, merge new values into existing
+			m.DeepMerge(vals)
+		}
+	}
+	return m, nil
+}
+
+func parseLocalFile(ctx context.Context, path string) (Values, error) {
+	m := make(Values)
+
+	// Handle files
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer func(f *os.File) {
+		if closeErr := f.Close(); closeErr != nil {
+			// Log close errors, don't fail on them for read operations
+			logger.From(ctx).Warn("failed to close file", "path", path, "error", closeErr)
+		}
+	}(f)
+
+	// Decode and merge values
+	if err = yaml.NewDecoder(f).DecodeContext(ctx, &m); err != nil {
+		if errors.Is(err, io.EOF) {
+			return m, nil // Empty file is ok
+		}
+		return nil, &YAMLDecodeError{
+			FilePath: path,
+			Err:      fmt.Errorf("%s", yaml.FormatError(err, true, true)),
+		}
+	}
+	return m, nil
+}
+
+// CheckSchema_Stub is intended to take a JSON schema and validate it against the values file(s).
+// TODO: implement public
+// TODO: Some open design questions:
+// - Do we take a json or byte array, a map[string]any, or a specific json.schema type?
+// - Do we want to return a list of errors, some specific schema fail datatype, or some other type?
+// - Surely there's libraries for this which have their own opinionated inputs for the schema and return types
+// func checkSchemaStub(_ Values, _ string) []error {
+// 	return nil
+// }
+
+// DeepMerge merges one or more Values maps recursively into the receiver via mutation.
+// Later maps in the variadic arguments take precedence over earlier ones.
+func (v Values) DeepMerge(sources ...Values) {
```

Comment:
++ this replicates the Helm behaviors well to my knowledge. (IE slice replacement behaviors)